omnigibson/utils/usd_utils.py

```diff
@@ -248,12 +249,93 @@ class BoundingBoxAPI:
+                - 3-array: end (x,y,z) corner of world-coordinate frame aligned bounding box
+        """
+        # Use the correct API to calculate AABB based on whether flatcache is enabled or not
+        return cls._compute_flatcache_aabb(prim=prim) if gm.ENABLE_FLATCACHE else \
+            cls._compute_non_flatcache_aabb(prim_path=prim.prim_path)
 
     @classmethod
-    def compute_aabb(cls, prim_path):
+    def _compute_flatcache_aabb(cls, prim):
         """
-        Computes the AABB (world-frame oriented) for the prim specified at @prim_path
+        Computes the AABB (world-frame oriented) for @prim. This an API compatible with flatcache, which manually
+        calculates the global transforms from the native bounding box to the updated values
+
+        NOTE: If @prim is an EntityPrim (i.e.: owns multiple links), then the computed bounding box will be
+        the subsequent aggregate over all the links.
+
+        Args:
+            prim (XFormPrim): Prim to calculate AABB for
+
+        Returns:
+            2-tuple:
+                - 3-array: start (x,y,z) corner of world-coordinate frame aligned bounding box
+                - 3-array: end (x,y,z) corner of world-coordinate frame aligned bounding box
+        """
+        # Run imports here to avoid circular imports
+        from omnigibson.prims import EntityPrim, RigidPrim, XFormPrim
+
+        # Simply grab the AABB if it's already been cached
+        if prim in cls.CACHE_FLATCACHE:
+            return cls.CACHE_FLATCACHE[prim]
+
+        # Next, process the AABB depending on the type of prim it is
+        if isinstance(prim, EntityPrim):
+            # Aggregate all owned points from entity prim
+            points = np.array([cls.compute_aabb(prim=link) for link in prim.links.values() if len(link.visual_meshes) > 0])  # Shape (N, 2, 3)
+            val = np.min(points[:, 0, :], axis=0), np.max(points[:, 1, :], axis=0)
+
+        elif isinstance(prim, XFormPrim):
+            # Simply calculate for this exact prim
+            # We assume the world transform from the USD is accurate at the prim's initial pose
+            val = cls._compute_non_flatcache_aabb(prim_path=prim.prim_path)
+
+            # If the simulation is not stopped and we're a rigid prim, then we need to manually compute the real-time
+            # transform from the prim's initial pose to the current pose
+            if isinstance(prim, RigidPrim) and not og.sim.is_stopped():
+                val = np.stack(
+                    [arr.flatten() for arr in np.meshgrid(*(np.linspace(lo, hi, 2) for lo, hi in zip(*val)))] + [
+                        np.ones(8, )])
+                # tf = T.pose2mat(prim.get_position_orientation()) @ T.pose2mat(prim.spawn_position_orientation)
+                tf = T.pose_inv(T.pose2mat(prim.spawn_position_orientation)) @ T.pose2mat(prim.get_position_orientation())
+                val = (tf @ val)[:-1, :]
+                val = np.min(val, axis=1), np.max(val, axis=1)
```

Comment:
sorry I am not sure I follow what is happening here

Reply:
resolved